caput/pipeline.py

```diff
@@ -967,6 +1015,23 @@ def _setup_keys(self, in_=None, out=None, requires=None):
         self._in = [queue.Queue() for i in range(n_in)]
         self._out_keys = out
 
+    @property
+    def out_keys(self):
+        """Get names of outputs of this task."""
+        return self._out_keys
+
+    @property
+    def pipeline_state(self):
+        """
+        Get state of the task.
+
+        Returns
+        -------
+        pipeline_state : str
+            One of 'setup', 'next', 'finish' or 'raise'.
+        """
+        return self._pipeline_state
+
```

Comment:
Why bother? These are only used internally within the pipeline framework. I don't think we want them to be exposed to a public level.